sdk-node/src/workflows/workflow.ts

```diff
@@ -252,186 +290,189 @@ export class Workflow<TInput extends WorkflowInput, name extends string> {
       },
     });
 
-    return {
-      ...jobCtx,
-      llm: l1m,
-      result: async <TResult>(
-        name: string,
-        fn: (ctx: WorkflowContext<TInput>) => Promise<TResult>,
-      ): Promise<TResult> => {
-        const ctx = this.createWorkflowContext(
-          version,
-          executionId,
-          input,
-          jobCtx,
-          clusterId,
-        );
+    const memo = async <TResult>(
+      name: string,
+      fn: (ctx: WorkflowContext<TInput>) => Promise<TResult>,
+    ): Promise<TResult> => {
+      const ctx = this.createWorkflowContext(
+        version,
+        executionId,
+        input,
+        jobCtx,
+        clusterId,
+      );
+
+      const serialize = (value: unknown) => JSON.stringify({ value });
+      const deserialize = (value: string) => {
+        try {
+          return JSON.parse(value).value;
+        } catch {
+          return null;
+        }
+      };
 
-        const serialize = (value: unknown) => JSON.stringify({ value });
-        const deserialize = (value: string) => {
-          try {
-            return JSON.parse(value).value;
-          } catch {
-            return null;
-          }
-        };
+      const existingValue = await this.client.getClusterKV({
+        params: {
+          clusterId: await this.getClusterId(),
+          key: `${executionId}_result_${name}`,
+        },
+      });
 
-        const existingValue = await this.client.getClusterKV({
-          params: {
-            clusterId: await this.getClusterId(),
-            key: `${executionId}_result_${name}`,
-          },
+      if (existingValue.status === 200) {
+        const existingValueParsed = deserialize(existingValue.body.value);
+
+        if (existingValueParsed) {
+          return existingValueParsed;
+        }
+      }
+
+      const result = await fn(ctx);
+
+      // TODO: async/retry
+      const setResult = await this.client.setClusterKV({
+        params: {
+          clusterId: await this.getClusterId(),
+          key: `${executionId}_result_${name}`,
```

Comment:
should be able to rename this to `_memo_` as well, will break the UI in some older runs which is probably fine for now?